bool r
bool g
bool b
---
int32  success